stdlib/asm/str/rlp.masm

```diff
@@ -0,0 +1,128 @@
+# Total Number of VM Cycles: 638 #
+# This is used for 5 field-elements -> 4 field-elements RLP converting. #
+# Used to decode a RLP code which contains up to 34 chars, the decoding result is ASCII #
+# input: [a, b_0, b_1, b_2, b_3, b_4] #
+#   a: a_hi -- whether the decoding result will be put into memory #
+#      a_lo -- only if a_hi is '1', this is the memory address to put the decoding result in  #
+#   b_0 to b_4 are the RLP code #
```

Comment:
I wonder if for this procedure we should not bother with memory at all. Basically, the stack transition would be just:
```
[b_4, b_3, b_2, b_1, b_0] -> [a, c_3, c_2, c_1, c_0]
```